Improve sedov and evrard test runs

jaescartin1 opened the pull request:
- Sedov cosmetic changes
- Evrard some fixes to run. Add the initial input file (Test3DEvrardRel.bin)
- Remove some warnings

jaescartin1 committed:
Ignore '-Wunknown-pragmas' in OpenOMP pragmas inside unit-test

jaescartin1 committed:
Skip 'OpenMPI' warnings implementation:

In file included from
/usr/lib/x86_64-linux-gnu/openmpi/include/openmpi/ompi/mpi/cxx/mpicxx.h:277,
                 from /usr/lib/x86_64-linux-gnu/openmpi/include/mpi.h:2887,
                 from domain/include/cstone/primitives/mpi_wrappers.hpp:34,
                 from domain/include/cstone/domain/domaindecomp_mpi.hpp:36,
                 from domain/include/cstone/domain/domain.hpp:37,
                 from src/sedov/sedov.cpp:11:
/usr/lib/x86_64-linux-gnu/openmpi/include/openmpi/ompi/mpi/cxx/op_inln.h:
In member function ‘virtual void MPI::Op::Init(void (*)(const void*,
void*, int, const MPI::Datatype&), bool)’:
/usr/lib/x86_64-linux-gnu/openmpi/include/openmpi/ompi/mpi/cxx/op_inln.h:121:25:
warning: cast between incompatible function types from ‘void (*)(void*,
void*, int*, ompi_datatype_t**, void (*)(void*, void*, int*,
ompi_datatype_t**))’ to ‘void (*)(void*, void*, int*,
ompi_datatype_t**)’ [-Wcast-function-type]
  121 |     (void)MPI_Op_create((MPI_User_function*)
ompi_mpi_cxx_op_intercept,
      |                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/lib/x86_64-linux-gnu/openmpi/include/openmpi/ompi/mpi/cxx/op_inln.h:123:38:
warning: cast between incompatible function types from ‘void (*)(const
void*, void*, int, const MPI::Datatype&)’ to ‘void (*)(void*, void*,
int*, ompi_datatype_t**)’ [-Wcast-function-type]
  123 |     ompi_op_set_cxx_callback(mpi_op, (MPI_User_function*) func);
      |                                      ^~~~~~~~~~~~~~~~~~~~~~~~~

jaescartin1 committed:
Added flag '-Wno-deprecated-gpu-targets' to NVCCLDFLAGS variable as well

jaescartin1 committed:
Split TESTCASE from the TESTCODE file. Needed for evard_new.cpp

jaescartin1 committed:
First evrard try: Typo Fixed

jaescartin1 committed:
Some fixes in order to execute evrard test

jaescartin1 committed:
Remove $(HPP)

jaescartin1 committed:
.

jaescartin1 committed:
Changed name

jaescartin1 committed:
Move input file for evrard

jaescartin1 committed:
Remove redundant namespaces

jaescartin1 committed:
Add default values to the help

jaescartin1 committed:
Fix typo in help

jaescartin1 committed:
Improving help

jaescartin1 committed:
Remove the old evrard.cpp because is not compatible with the new code.

jaescartin1 committed:
Change default file in the evrard test

jgphpc commented:
remove this Test3DEvrardRel.bin please

sekelle commented on src/evrard/evrard_new.cpp:
```diff
@@ -42,13 +47,13 @@ int main(int argc, char** argv)
     const bool quiet = parser.exists("--quiet");
     const std::string checkpointInput = parser.getString("--cinput");
-    const std::string inputFilePath = parser.getString("--input", "bigfiles/Test3DEvrardRel.bin");
+    const std::string inputFilePath = parser.getString("--input", "../../../src/evrard/bigfiles/Test3DEvrardRel.bin");
```
we could change this to
```suggestion
    const std::string inputFilePath = parser.getString("--input", "./Test3DEvrardRel.bin");
```

sekelle commented:
Please remove the large binary file from the PR. The rest looks fine.

jaescartin1 committed:
Update src/evrard/evrard_new.cpp

Co-authored-by: Sebastian Keller <44800242+sebkelle1@users.noreply.github.com>

jaescartin1 committed:
Some changes:

- Removed 'src/evrard/bigfiles/Test3DEvrardRel.bin'
- Changed 'evrard_new.cpp' to 'evrard.cpp' and subsequence files
- Changed default test to Sedov in Makefile

jaescartin1 commented:
I removed the initial file './Test3DEvrardRel.bin'. Although I put this file as input, the evrard execution shows some errors. 
Is it possible to execute without this initial file? 
Please, How do you do?

sekelle commented on domain/test/unit/CMakeLists.txt:
```diff
@@ -28,7 +28,7 @@ set(UNIT_TESTS
 
 add_executable(component_units ${UNIT_TESTS})
-target_compile_options(component_units PRIVATE -Wall -Wextra)
+target_compile_options(component_units PRIVATE -Wall -Wextra -Wno-unknown-pragmas)
```
No longer needed, this has just been merged into develop.

jaescartin1 committed:
Rolled back 'domain/test/unit' to the version in branch develop

sekelle commented:
It's ok to rename the old evrard.cpp file, but the renaming should be done in a follow-up PR. Because if you both introduce changes in evrard_new.cpp and rename the file, the diff will be obscured.
So in this PR, you can delete evrard.cpp, but do the renaming of evrard_new to evrard in a follow up.
This will also make it easier for @jgphpc to review the CI changes.

sekelle commented:
> I removed the initial file './Test3DEvrardRel.bin'. Although I put this file as input, the evrard execution shows some errors. Is it possible to execute without this initial file? Please, How do you do?

The file is needed as input. But you could also provide any other file with valid initial conditions, with any number of particles.
Note that running with more than 1 rank is not enabled yet for this test case.

I'll update the README.md file to describe how the binary file can be generated from the .txt with 3 lines of python.

jaescartin1 commented:
Perfect, thanks. I need to provide the analytical solution for evrard test and I need to know how to execute properly the test in the code.

I just see that is difficult to read the changes removing and renaming the old cpp of the evrard test. In the review code, the file is mixed with the previous one ... and I understood that I will be difficult to verify. As you suggested, next time I will do this kind of changes in two steps.

sekelle committed:
Merge remote-tracking branch 'origin' into test-sedov

sekelle committed:
updated default input file path in help message
